### PR TITLE
Improved NVTX documentation

### PR DESCRIPTION
**Pull request by rietmann-nv**
Some users working with CuPy did not understand NVTX ranges, so I tried to improve the docs with a small example and a sentence about using Nsight Systems. Instead of complaining about the docs I thought I would spend a minute and contribute. 

I don't know if I got the formatting correct for the "Example:" code in the docstring.

### CODE REVIEW

**Comment by leofang** [comment]
There are flake8 errors, please check.

> I don't know if I got the formatting correct for the "Example:" code in the docstring.

One of the CIs is using Read the Docs (RTD) to build docs, you can check how it renders upon merge 🙂

File: cupy_backends/cuda/libs/nvtx.pyx [hunk 1]
```diff
@@ -144,9 +144,26 @@ cpdef RangePushC(message, uint32_t color=0):
     """
     Starts a nested range.
 
+    Ranges are used to describe events over a time span during execution of the
+    application. This is particularly useful when profiling with Nsight Systems
+    to help connect user-specified ranges with cupy-internal CUDA-kernels. The
+    duration of a range is defined by the corresponding pair of
```
Comment (leofang) [inline cupy_backends/cuda/libs/nvtx.pyx:150]: nit:
```suggestion
    to help connect user-specified ranges with CuPy's internal CUDA-kernels.
    The duration of a range is defined by the corresponding pair of
```

File: cupy_backends/cuda/libs/nvtx.pyx [hunk 2]
```diff
@@ -173,9 +190,22 @@ cpdef RangePush(message, int id_color=-1):
...
     Starts a nested range.
 
-    Ranges are used to describe events over a time span during execution of
-    the application. The duration of a range is defined by the corresponding
-    pair of ``RangePush*()`` to ``RangePop()`` calls.
+    Ranges are used to describe events over a time span during execution of the
+    application. This is particularly useful when profiling with Nsight Systems
+    to help connect user-specified ranges with cupy-internal CUDA-kernels. The
+    duration of a range is defined by the corresponding pair of ``RangePush()``
+    to ``RangePop()`` calls, which can be nested.
```
Comment (leofang) [inline cupy_backends/cuda/libs/nvtx.pyx:197]: Same nitpick as above.

File: cupy_backends/cuda/libs/nvtx.pyx [hunk 3]
```diff
@@ -144,9 +144,26 @@ cpdef RangePushC(message, uint32_t color=0):
...
+        for i in range(N):
+            RangePushC("Iter {}: Double A".format(i))
+            A = 2*A
+            RangePop()
+        RangePop()
+
     Ranges are used to describe events over a time span during execution of
     the application. The duration of a range is defined by the corresponding
-    pair of ``RangePush*()`` to ``RangePop()`` calls.
+    pair of ``RangePushC()`` to ``RangePop()`` calls.
```
[the remark this answers is not in the document]
Comment (rietmann-nv) [reply cupy_backends/cuda/libs/nvtx.pyx:166]: Yes makes sense to me.

File: cupy_backends/cuda/libs/nvtx.pyx [hunk 4]
```diff
@@ -144,9 +144,22 @@ cpdef RangePushC(message, uint32_t color=0):
...
-    Ranges are used to describe events over a time span during execution of
-    the application. The duration of a range is defined by the corresponding
-    pair of ``RangePush*()`` to ``RangePop()`` calls.
+    Ranges are used to describe events over a time span during execution of the
+    application. This is particularly useful when profiling with Nsight Systems
+    to help connect user-specified ranges with CuPy's internal CUDA-kernels.
+    The duration of a range is defined by the corresponding pair of
+    ``RangePushC()`` to ``RangePop()`` calls, which can be nested.
+
+    Example:
```
Comment (leofang) [inline cupy_backends/cuda/libs/nvtx.pyx:153]: I could be wrong, but I guess Sphinx expects the `Examples` section to have the Python prompts. How about trying this instead?
```suggestion
    .. code-block:: python
```

File: cupy_backends/cuda/libs/nvtx.pyx [hunk 5]
```diff
@@ -173,9 +186,22 @@ cpdef RangePush(message, int id_color=-1):
...
-    Ranges are used to describe events over a time span during execution of
-    the application. The duration of a range is defined by the corresponding
-    pair of ``RangePush*()`` to ``RangePop()`` calls.
+    Ranges are used to describe events over a time span during execution of the
+    application. This is particularly useful when profiling with Nsight Systems
+    to help connect user-specified ranges with CuPy's internal CUDA-kernels.
+    The duration of a range is defined by the corresponding pair of
+    ``RangePush()`` to ``RangePop()`` calls, which can be nested.
+
+    Example:
```
Comment (leofang) [inline cupy_backends/cuda/libs/nvtx.pyx:195]:
```suggestion
    .. code-block:: python
```

**Comment by kmaehashi** [comment]
Thanks, the docs seems rendered nicely :)
https://cupy--6774.org.readthedocs.build/en/6774/reference/generated/cupy.cuda.nvtx.RangePush.html

/test mini

**Review by emcastillo — APPROVED** [review]
Thanks!!!

**Pull request merged**